File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Peril.java
package com.umbrella.insurance.core.models.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "perils", schema = "public")
public class Peril {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "peril_id", nullable = false)
    private Long id;

    @Column(name = "peril_name", length = Integer.MAX_VALUE)
    private String perilName;

    @Column(name = "description", length = Integer.MAX_VALUE)
    private String description;

    @Column(name = "scale_name", length = Integer.MAX_VALUE)
    private String scaleName;

    @NotNull
    @Column(name = "min_magnitude", nullable = false)
    private Double minMagnitude;

    @NotNull
    @Column(name = "max_magnitude", nullable = false)
    private Double maxMagnitude;

}

File: umbrella-insurance-react-ui/src/models/perils/v1/Peril.ts
export class Peril {
    id?: number;
    perilName?: string;
    description?: string;
    scaleName?: string;
    minMagnitude?: number;
    maxMagnitude?: number;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.perilName) {
                this.perilName = obj.perilName;
            }
            if(obj?.description) {
                this.description = obj.description;
            }
            if(obj?.scaleName) {
                this.scaleName = obj.scaleName;
            }
            if(obj?.minMagnitude) {
                this.minMagnitude = obj.minMagnitude;
            }
            if(obj?.maxMagnitude) {
                this.maxMagnitude = obj.maxMagnitude;
            }
        }
    }
}